### PR TITLE
MODORGS-12 Improve Unit test Execution

### PR DESCRIPTION
**Pull request by pautau-s**
## Purpose
[MODORGS-12](https://issues.folio.org/browse/MODORGS-12) Improve Unit test execution

## Approach

* added test suite that allows
  * Creation of postgres database and a tenant is done just once prior to all the Test classes
  * Deletion of the tenant and closing the database is done after all the Test classes
* used parameterized test for identical test scenarios to avoid duplicating code
* extracted common logic for getting list of entities to `HelperUtils.getEntitiesCollection` method

##Additional changes
* `publishModDescriptor` and `mvnDeploy` changed to 'yes' in Jenkinsfile
* changed interface id in moduleDescriptor for url from singular to plural form in order to be consistent with other interfaces
* pointed acq-models to merge-commit with changes from [folio-org/acq-models#116](https://github.com/folio-org/acq-models/pull/116)

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were there any schema changes?
  - [x] There are no breaking changes in this PR.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 buildMvn {
-  publishModDescriptor = 'no'
+  publishModDescriptor = 'yes'
```
Comment (craigmcnally) [inline Jenkinsfile]: are these changes intentional?  I didn't see them mentioned in the PR description

File: descriptors/ModuleDescriptor-template.json [hunk 2]
```diff
@@ -189,7 +189,7 @@
       ]
     },
     {
-      "id": "organizations-storage.url",
+      "id": "organizations-storage.urls",
```
Comment (craigmcnally) [inline descriptors/ModuleDescriptor-template.json]: I'm fine with this change, but a little surprised to see this here since it isn't mentioned in the description...

File: Jenkinsfile (hunk 1 above)
Comment (pautau-s) [reply Jenkinsfile]: PR description updated

File: descriptors/ModuleDescriptor-template.json (hunk 2 above)
Comment (pautau-s) [reply descriptors/ModuleDescriptor-template.json]: PR description updated

File: src/test/java/org/folio/rest/utils/TestEntities.java [hunk 3]
```diff
@@ -1,11 +1,23 @@
...
   CONTACT("/organization-storage/contacts", Contact.class, "contact.sample",  "notes", "ABC123", 13),
-  ORGANIZATION("/organization-storage/organizations", Organization.class, "organization.sample",  "code", "ABC123", 13);
+  EMAIL("/organization-storage/emails", Email.class, "email.sample", "value", "test@folio.org", 0),
+  INTERFACE("/organization-storage/interfaces", Interface.class, "interface.sample", "name", "Test Portal", 0),
+  ORGANIZATION("/organization-storage/organizations", Organization.class, "organization.sample",  "code", "ABC123", 13),
+  PHONE_NUMBER("/organization-storage/phone-numbers", PhoneNumber.class, "phoneNumber.sample", "phoneNumber", "9999999999", 0),
+  URL("/organization-storage/urls", Url.class, "url.sample", "value", "http://test.org", 0);
 
 
   TestEntities(String endpoint, Class<?> clazz, String sampleFileName, String updatedFieldName, String updatedFieldValue, int initialQuantity) {
```
[the remark this answers is not in the document]
Comment (pautau-s) [reply src/test/java/org/folio/rest/utils/TestEntities.java]: Done

File: src/main/java/org/folio/rest/impl/AddressesAPI.java [hunk 4]
```diff
@@ -1,32 +1,26 @@
...
+import org.folio.rest.persist.QueryHolder;
 
 import javax.ws.rs.core.Response;
-import java.util.List;
 import java.util.Map;
 
+import static org.folio.rest.persist.HelperUtils.getEntitiesCollection;
+
 public class AddressesAPI implements OrganizationStorageAddresses {
   private static final String ADDRESS_TABLE = "address";
```
[the remark this answers is not in the document]
Comment (pautau-s) [reply src/main/java/org/folio/rest/impl/AddressesAPI.java]: Renamed the tables
